### PR TITLE
Issue 1 Create an Entry

### PR DESCRIPTION
**Pull request by AddyAbid**
https://user-images.githubusercontent.com/85080021/132600056-db38f6f3-a5a9-436b-b8ea-f368c6583197.mp4

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -16,6 +19,63 @@
...
+            name="notes"
+            type="text"
+            id="notes"></textarea>
+
+              <button type="submit">SAVE</button>
+
+          </div>
+      </form>
+    </div>
+    </div>
```
Comment (codingcodymiller) [inline index.html:78]: Your indentation seemed to get kinda messed up here, please revise. Doubler check that you ran `npm ci` in your repo, too

File: index.html [hunk 2]
```diff
@@ -16,6 +19,63 @@
...
   <header>
+    <div class="nav-container ">
+      <div class="row">
+        <div class="column-full">
+          <div class="nav container">
+            <h3>Code Journal</h3>
+          </div>
+        </div>
+      </div>
+    </div>
```
Comment (codingcodymiller) [inline index.html:30]: Only this portion should be a child of the `header` element, the rest of the HTML for the form content should be in the `main` element.

File: js/main.js [hunk 3]
```diff
@@ -1,2 +1,55 @@
 /* global data */
 /* exported data */
+var $imgInput = document.querySelector('#photo-url');
+
+var $imgSource = document.querySelector('img');
+var placeholderImagePath = 'images/placeholder-image-square.jpg';
+$imgInput.addEventListener('input', changeImage);
+
+function changeImage(event) {
+  if (event.target) {
```
Comment (codingcodymiller) [inline js/main.js:10]: Why `if(event.target)`? If this is an event handler, that should always have a truth value. What you really want to check is if the input for the URL has a value. If there is no value inside it, you should set the `src` to the placeholder image, but if there is a value, use that as the new `src`

File: js/main.js [hunk 4]
```diff
@@ -1,2 +1,55 @@
...
+    notes: notes
+  };
+
+  formValues.entryId = data.nextEntryId;
+
+  data.nextEntryId++;
+
+  data.entries.push(formValues);
+
+  $imgSource.src = placeholderImagePath;
```
Comment (codingcodymiller) [inline js/main.js:41]: I would use `setAttribute` rather than assigning directly to the `src` property like this.

File: js/main.js [hunk 5]
```diff
@@ -1,2 +1,55 @@
...
+  $entryForm.reset();
+
+  return formValues;
```
Comment (codingcodymiller) [inline js/main.js:44]: There won't be any `return` necessary here.

File: js/main.js [hunk 6]
```diff
@@ -1,2 +1,55 @@
...
+
+var lastInput = localStorage.getItem('data-entries');
+if (lastInput !== null) {
+  data.entries = JSON.parse(lastInput);
+}
+window.addEventListener('beforeunload', function (event) {
+  var dataJSON = JSON.stringify(data);
+  localStorage.setItem('entries-data', dataJSON);
+
+});
```
Comment (codingcodymiller) [inline js/main.js:55]: This code should be in `data.js`. You want the localStorage code to be in the same file as the data model it is preserving.

File: js/main.js [hunk 7]
```diff
@@ -1,2 +1,55 @@
...
+
+  $imgSource.src = placeholderImagePath;
+  $entryForm.reset();
+
+  return formValues;
+}
+
+var lastInput = localStorage.getItem('data-entries');
+if (lastInput !== null) {
+  data.entries = JSON.parse(lastInput);
```
Comment (codingcodymiller) [inline js/main.js:49]: This is incorrect, you should be replacing the value of `data` entirely with the object you parsed from `localStorage`.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by AddyAbid** [comment]
https://user-images.githubusercontent.com/85080021/132602463-d40ea4d2-f691-4ba0-b527-ccec8bcd416d.mp4

File: js/main.js [hunk 8]
```diff
@@ -1,2 +1,44 @@
 /* global data */
 /* exported data */
+var $imgInput = document.querySelector('#photo-url');
+
+var $imgSource = document.querySelector('img');
+var placeholderImagePath = 'images/placeholder-image-square.jpg';
+$imgInput.addEventListener('input', changeImage);
+
+function changeImage(event) {
+  if ($imgInput !== '') {
```
Comment (codingcodymiller) [inline js/main.js:10]: `$imgInput` is the element itself, it will never be an empty string. You need to check `$imgInput.value`

File: js/main.js [hunk 9]
```diff
@@ -1,2 +1,44 @@
...
+    $imgSource.src = $imgInput.value;
+  } else {
+    $imgSource.src = placeholderImagePath;
+  }
+  return changeImage;
```
Comment (codingcodymiller) [inline js/main.js:15]: No return is necessary, this is an event handler.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by AddyAbid** [comment]
https://user-images.githubusercontent.com/85080021/132603020-44417e53-37af-48b2-82bf-f982d1c25a47.mp4

**Review by codingcodymiller — APPROVED** [review]

**Pull request merged**